framework/source/class/qx/ui/dialog/Abstract.js

```diff
@@ -0,0 +1,207 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2016 Yeshua Rodas, http://yybalam.net
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Yeshua Rodas (yybalam)
+
+************************************************************************ */
+
+/**
+ * This is the base class for dialogs.
+ *
+ * For this class a dialog has been thinking as is a modal window
+ * with a title, a message, an icon and some action button(s)
+ * and without the captionbar elements.
+ */
+qx.Class.define('qx.ui.dialog.Abstract', {
+  extend : qx.ui.window.Window,
+  type : "abstract",
+
+  /**
+   * @param title {String?null} Title of dialog.
+   * @param message {String?null} Message to show.
+   * @param icon {String?null} Icon to use.
+   */
+  construct : function(title, message, icon) {
+    title = title || '';
+    message = message || '';
+
+    this.base(arguments);
+
+    this.setLayout(new qx.ui.layout.VBox());
+
+    this.setMaxWidth(320);
+    this.setResizable(false);
+
+    this.add(this._getAtom());
+    this.add(this._getButtonsBar());
+
+    this._initDialog(title, message);
+
+    if (icon != undefined) {
+      this._getAtom().setIcon(icon);
+    }
+  },
```

Comment:
you may want to follow the structure outlined [in the qooxdoo manuel](http://www.qooxdoo.org/current/pages/desktop/ui_develop.html) using ChildControls